--- su.c.orig	Sun Jun 30 10:56:57 2002
+++ su.c	Sun Jun 30 10:56:02 2002
@@ -509,6 +509,9 @@
       char *arg0;
       char *shell_basename;
 
+      if (chdir (pw->pw_dir))
+        error (0, errno, _("warning: cannot change directory to %s"), 
+	       pw->pw_dir);
       shell_basename = base_name (shell);
       arg0 = xmalloc (strlen (shell_basename) + 2);
       arg0[0] = '-';
@@ -762,8 +765,6 @@
 #else
   change_identity (pw);
 #endif
-  if (simulate_login && chdir (pw->pw_dir))
-    error (0, errno, _("warning: cannot change directory to %s"), pw->pw_dir);
 
   run_shell (shell, command, additional_args, pw);
 }
